packages/web3-eth-abi/src/api/logs_api.ts

```diff
@@ -0,0 +1,62 @@
+import { HexString } from 'web3-utils';
+import { JsonAbiParameter } from '../types';
+import { decodeParameter, decodeParametersWith } from './parameters_api';
+
+const STATIC_TYPES = ['bool', 'string', 'int', 'uint', 'address', 'fixed', 'ufixed'];
+
+/**
+ * Decodes events non- and indexed parameters.
+ */
+export const decodeLog = <ReturnType extends Record<string, unknown>>(
+	inputs: Array<JsonAbiParameter>,
+	data: HexString,
+	topics: string | string[],
+) => {
+	const clonedTopics = Array.isArray(topics) ? topics : [topics];
```

Comment:
We might want to add validation to these methods (after this PR is merged) using methods such as [isTopic](https://github.com/ChainSafe/web3.js/blob/e89b6a0fde385a8009891ec0871bb2c55ce6827f/packages/web3-utils/src/validation.ts#L381). We might also want to create a [isTopics](https://github.com/ChainSafe/web3.js/pull/4525/files#diff-170228df6bd950804f5f0071be007d1f2acba8a41ef2f9959b815e6ca82a1d97R480) methods that handles an array of topics

Reply:
Will look at it explicitly in upcoming PR.